v2/dtos/requests/event.go

```diff
@@ -34,25 +32,30 @@ type UpdateEventPushedByChecksumRequest struct {
 
 // Validate satisfies the Validator interface
 func (a AddEventRequest) Validate() error {
-	err := v2.Validate(a)
-	return err
+	if err := v2.Validate(a); err != nil {
+		return err
+	}
+
+	// validate the nested BinaryReading and SimpleReading struct
+	for _, r := range a.Event.Readings {
```

Comment:
Yes, originally it would actually check the nested structs with the following annotation added:
https://github.com/edgexfoundry/go-mod-core-contracts/pull/264/files/a67534a7810fd718be6ef5ae376fb7cd36490ba5#diff-7a43efa7e1bef0ec66b9fa81dddddf11R24

However, as we only have either `SimpleReading` or `BinaryReading` in `BaseReading` DTO, the above check would cause the validation failed so the we do the following to skip the validation of `SimpleReading` and `BinaryReading`:
https://github.com/edgexfoundry/go-mod-core-contracts/pull/264/files/a67534a7810fd718be6ef5ae376fb7cd36490ba5#diff-badcd41a51a415996619cb57beb0c5c5R33-R34

Therefore additional handling is added to handle the nested validation of `SimpleReading` and `BinaryReading`:
https://github.com/edgexfoundry/go-mod-core-contracts/pull/264/files/a67534a7810fd718be6ef5ae376fb7cd36490ba5#diff-fd093d7d5d89b1cdab65572c38e1b4baR39-R42

Would you please suggest if any other workaround you think of?? Thanks.

Reply:
Ok, got it, if it is a `SimpleReading` the validation fails for the `BinaryReading` when using annotation.
Please comment `why doing this` in the code. THX!

Reply:
Comments added, thanks.